src/robotFactory.js

```diff
@@ -1,10 +1,122 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(
+    name,
+    weight,
+    coords,
+    chipVersion
+  ) {
+    this.name = name;
+    this.weight = weight;
+    this.chipVersion = chipVersion;
 
-class FlyingRobot {}
+    this.coords = coords.x ? coords : {
+      x: 0,
+      y: 0,
+    };
+  }
 
-class DeliveryDrone {}
+  goForward(step) {
+    if (step) {
+      this.coords.y += step;
+    } else {
+      this.coords.y++;
+    }
+  };
```

Comment:
```suggestion
  goForward(step = 1) {
    this.coords.y += step;
  }
```
Просто установить по умолчанию значение, и так же в следующих случаях